x-pack/plugins/upgrade_assistant/README.md

```diff
@@ -2,66 +2,190 @@
     * Delete the old index and rename the prefixed index.
-  * Some apps might require custom scripts, as was the case with APM ([#29845](https://github.com/elastic/kibana/pull/29845)).
-    In that case the migration performed a reindex with a Painless script (covered by automated tests)
-    that made the required changes to the data.
-* **Update index settings.** Some index settings will need to be updated, which doesn't require a
-reindex. An example of this is the "Fix" button that was added for metricbeat and filebeat indices
-([#32829](https://github.com/elastic/kibana/pull/32829), [#33439](https://github.com/elastic/kibana/pull/33439)).
+* **Updating index settings.** Some index settings will need to be updated, which doesn't require a
+reindex. An example of this is the "Remove deprecated settings" button, which is shown when [deprecated translog settings](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-translog.html#index-modules-translog-retention) are detected ([#93293](https://github.com/elastic/kibana/pull/93293)).
+* **Upgrading or deleting snapshots**. This is specific to Machine Learning. If a user has old Machine Learning job model snapshots, they will need to be upgraded or deleted. The Upgrade Assistant provides a way to resolve this automatically for the user ([#100066](https://github.com/elastic/kibana/pull/100066)).
 * **Following the docs.** The Deprecation Info API provides links to the deprecation docs. Users
 will follow these docs to address the problem and make these warnings or errors disappear in the
 Upgrade Assistant.
-* **Stopping/restarting tasks and jobs.** Users had to stop watches and ML jobs and restart them as
-soon as reindexing was complete ([#29663](https://github.com/elastic/kibana/pull/29663)).
\ No newline at end of file
+
+#### Kibana
+
+Kibana deprecations can be handled in one of two ways:
+
+* **Automatic resolution.** Some deprecations can be fixed automatically through Upgrade Assistant via an API call. When this is possible, users will see a "Quick resolve" button in the Upgrade Assistant.
+* **Manual steps.** For deprecations that require the user to address manually, the Upgrade Assistant provides a list of steps to follow as well as a link to documentation. Once the deprecation is addressed, it will no longer appear in the Upgrade Assistant.
+
+### Steps for testing
+#### Elasticsearch deprecations
+
+To test the Elasticsearch deprecations page ([#107053](https://github.com/elastic/kibana/pull/107053)), you will first need to create a set of deprecations that will be returned from the deprecation info API.
+
+**1. Reindexing**
+
+  The reindex action appears in UA whenever the deprecation `Index created before XX` is encountered. To reproduce, you will need to start up a cluster on the previous major version (e.g., if you are running 7.x, start a 6.8 cluster). Create a handful of indices, for example:
```

Comment:
It might be a time-saver to also provide a link to a zip file of 6.x data (perhaps stored in a gist), but not a blocker for this PR.